File: LivrusVirtus/Delphi7/CLX/dbExpress/fontes/ConEnderecos.pas
unit ConEnderecos;

interface

uses
  SysUtils, Types, Classes, Variants, QTypes, QGraphics, QControls, QForms, 
  QDialogs, QStdCtrls, Consultas, FMTBcd, Provider, DBClient, DB, SqlExpr,
  QGrids, QDBGrids, QExtCtrls;

type
  TfrmConEnderecos = class(TfrmConsultas)
    procedure FormClose(Sender: TObject; var Action: TCloseAction);
    procedure btnLimparClick(Sender: TObject);
    procedure btnPesquisarClick(Sender: TObject);
    procedure FormShow(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmConEnderecos: TfrmConEnderecos;

implementation

uses Principal, ConsultasSQL;

{$R *.xfm}

procedure TfrmConEnderecos.FormClose(Sender: TObject;
  var Action: TCloseAction);
begin
frmPrincipal.Enderecos2.Enabled:=true;
inherited;
end;

procedure TfrmConEnderecos.btnLimparClick(Sender: TObject);
begin
  inherited;
LimparDados(ConsSQL.Endereco('S','0','','','','',''));
end;

procedure TfrmConEnderecos.btnPesquisarClick(Sender: TObject);
begin
  inherited;
PesquisarDados(ConsSQL.Endereco('S','', edtDescricao.Text,'','','',''),
'Logradouro não encontrado!');
end;

procedure TfrmConEnderecos.FormShow(Sender: TObject);
begin
  inherited;
btnLimparClick(self);
end;

end.

File: LivrusVirtus/Kylix3/cpp/fontes/ConEnderecos.cpp
//---------------------------------------------------------------------------

#include <clx.h>
#pragma hdrstop

#include "ConEnderecos.h"
#include "ConsultasSQL.h"
#include "Principal.h"
//---------------------------------------------------------------------------
#pragma link "Consultas"
#pragma resource "*.xfm"
TfrmConEnderecos *frmConEnderecos;
//---------------------------------------------------------------------------
__fastcall TfrmConEnderecos::TfrmConEnderecos(TComponent* Owner)
        : TfrmConsultas(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEnderecos::FormClose(TObject *Sender,
      TCloseAction &Action)
{
frmPrincipal->Enderecos2->Enabled=true;
TfrmConsultas::FormClose(Sender, Action);
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEnderecos::btnLimparClick(TObject *Sender)
{
LimparDados(ConsSQL->Endereco('S',"0","","","","",""));
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEnderecos::btnPesquisarClick(TObject *Sender)
{
PesquisarDados(ConsSQL->Endereco('S',"", edtDescricao->Text.c_bstr(),"","","",""),
"Endereço não encontrado!");
}
//---------------------------------------------------------------------------
void __fastcall TfrmConEnderecos::FormShow(TObject *Sender)
{
btnLimparClick(this);
}
//---------------------------------------------------------------------------
